File: tps/entrada e saida/tp01/TP01Q11 - RECURSIVO - Palíndromo em Java/RECPalindromo.java
class RECPalindromo{
   public static boolean isFim(String s){
      return (s.length() == 3 && s.charAt(0) == 'F' && s.charAt(1) == 'I' && s.charAt(2) == 'M');
   }

   public static String isPalindromo (String s){
     return isPalindromo(s, 0, s.length()-1);
   }

   public static String isPalindromo (String s, int i, int size){
      String isP = "NAO";

      if(s.charAt(i) == s.charAt(size)){
        isP = isPalindromo(s, i+1, size-1);
        isP = "SIM";
      } else{
        isP = "NAO";
      }

      return isP;
   }

   public static void main (String[] args){
      String[] entrada = new String[1000];
      int numEntrada = 0;

      //Leitura da entrada padrao
      do {
         entrada[numEntrada] = MyIO.readLine();
      } while (isFim(entrada[numEntrada++]) == false);
      numEntrada--;   //Desconsiderar ultima linha contendo a palavra FIM

      //Para cada linha de entrada, gerando uma de saida boleana se e' ou nao palindromo
      for(int i = 0; i < numEntrada; i++){
         MyIO.println(isPalindromo(entrada[i]));
      }
   }
}

File: tps/entrada e saida/tp01/TP01Q12 - RECURSIVO - Palíndromo em C/RECPalindromo.cpp
#include<stdio.h>
#include<stdbool.h>
#include<string.h>

bool isPalindromo (char s[], int i, int a)
{
    bool is = true;

    if(a < (i/2)){
      
          if(s[a] != s[i-1-a]){
            is = false;
          } else{
            is = true;
            is = isPalindromo(s, i, a+1);
          }
      
    }

    return(is);   
}

bool isFim(char s[], int i)
{
    return (i == 3 && s[0] == 'F' && s[1] == 'I' && s[2] == 'M');
}

void main( )
{
    int size = 0;
    char s[1000];
    int i = 0;

    do
    {
        fgets(s, 1000, stdin);
        size = strlen(s)-1; 
        
        if (!isFim(s, size))
        {
            if (isPalindromo(s, size, 0))
            {
                printf("SIM\n");
            } 
            else
            {
                printf("NAO\n");
            } 
        }

    }while (!isFim(s, size));
 
}